### DISCUSSION TITLE
Bonsai crashes with Audio Playback

### DISCUSSION THREAD

**Discussion by PiezaKake** [bonsai-rx/bonsai#1353] [Q&A] [answered]
Sometimes, when an audio file is read/played, Bonsai crashes. The audio file is 45 min long and has a sampling rate of 160 kHz. I read it with **AudioReader** with the **BufferLength** set to 0. I play the audio file with **AudioPlayback** with the **SampleRate** set to 160 kHz and **State** to Playing. The message when Bonsai crashes is:
![image](https://github.com/bonsai-rx/bonsai/assets/129788515/46f63871-5159-4cfc-91d4-2d521811bfc4)

**Comment by bruno-f-cruz** [comment, marked as answer]
Have you tried to not load the full waveform into memory at once? Set the buffer size to a different value (eg 1600000)

**Reply by PiezaKake** [reply]
Thanks for this, looks like it worked. I had played with the buffer size, but had gone up to only 10 ms.
